pkg/plugins/policies/meshhttproute/api/v1alpha1/meshhttproute.go

```diff
@@ -103,10 +104,46 @@ type HeaderModifier struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// PreciseHostname is the fully qualified domain name of a network host. This
+// matches the RFC 1123 definition of a hostname with 1 notable exception that
+// numeric IP addresses are not allowed.
+//
+// Note that as per RFC1035 and RFC1123, a *label* must consist of lower case
+// alphanumeric characters or '-', and must start and end with an alphanumeric
+// character. No other punctuation is allowed.
+//
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+type PreciseHostname string
+
+// PortNumber defines a network port.
+//
+// +kubebuilder:validation:Minimum=1
+// +kubebuilder:validation:Maximum=65535
+type PortNumber int32
+
+type RequestRedirect struct {
+	Scheme   *string          `json:"scheme,omitempty"`
```

Comment:
can scheme be just any string? [RFC3986](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) says:

> Scheme names consist of a sequence of characters beginning with a
   letter and followed by any combination of letters, digits, plus
   ("+"), period ("."), or hyphen ("-").

On the other side don't see any validation in Envoy but maybe it would crash anyway.

Reply:
Ah this probably deserves the same logic as gateway API so I'll make it an enum HTTP/HTTPS